src/documentation/0000-node-fs-fd-use-case-read-method/article.md

```diff
@@ -0,0 +1,61 @@
+---
+title: Use case of file descriptor returned by fs.open() method in fs.read()
+description: "Use of read method in fs module to handle reading huge amount of data in chunks."
+authors: TanmayBanerjee-23rd
+section: Quick Start
+category: learn
+---
+
+## Introduction to fs module read method.
+
+Using `fs.read()` method we can read a file in a more controlled manner in comparision to `fs.readFile()` or `fs.readFileSync()`.
+
+## Working with fs.read() method.
+
+Since `fs.readFile()` and `fs.readFileSync()` reads the full content of the file in memory before returning the data or in other words before you access and start operating on the data that is read, the methods will make sure the entire file is read which in case of huge sized file will have a great performance impact in terms of memory usage and speed of execution of the program.
+
+So, we can manage the above situation by using `fs.read()` method which gives us more control over reading the file. How? We will see it below.
+
+`fs.read()` Asynchronously reads data from the file referenced by the supplied file descriptor returned by `fs.open()`. Since it's asynchronous in nature which will help not to block our call stack.
```

Comment:
```suggestion
`fs.read()` asynchronously reads data from the file referenced by the supplied file descriptor returned by `fs.open()`. Since it's asynchronous in nature which will help not to block our call stack.
```